WordPress/src/test/java/org/wordpress/android/ui/jetpack/backup/download/usecases/GetBackupDownloadStatusUseCaseTest.kt

```diff
@@ -169,6 +169,57 @@ class GetBackupDownloadStatusUseCaseTest : BaseUnitTest() {
                 assertThat(result).contains(progressStatus, completeStatus)
             }
 
+    @Test
+    fun `given max fetch retries exceeded, when backup status triggers, then return remote request failure`() =
+            coroutineScope.runBlockingTest {
+                whenever(activityLogStore.fetchBackupDownloadState(any()))
+                        .thenReturn(
+                            OnBackupDownloadStatusFetched(
+                                    BackupDownloadStatusError(GENERIC_ERROR),
+                                    FETCH_BACKUP_DOWNLOAD_STATE
+                            ))
+                            .thenReturn(
+                                    OnBackupDownloadStatusFetched(
+                                            BackupDownloadStatusError(GENERIC_ERROR),
+                                            FETCH_BACKUP_DOWNLOAD_STATE
+                            ))
+                            .thenReturn(
+                                    OnBackupDownloadStatusFetched(
+                                            BackupDownloadStatusError(GENERIC_ERROR),
+                                            FETCH_BACKUP_DOWNLOAD_STATE
+                            ))
+
+                val result = useCase.getBackupDownloadStatus(site, downloadId).toList()
```

Comment:
⚠️ 
1. `runBlockingTests` invokes `dispatcher.advanceUntilIdle()` internally, so it skips all delays.
2. `toList()` method on Flow suspends the current thread until the Flow completes -> the `advanceTimeBy` doesn't have any effect here.

We can discuss this during the call. It's quite tricky.

Reply:
Based on our discussions, commit 1983a4a
(1) Replaced `MainCoroutineScopeRemove` with `test`
(2) Removed `advanceTimeBy`, as it wasn't needed

Reply:
Nice, thanks!